src/mlpack/methods/ne/genome.hpp

```diff
@@ -0,0 +1,334 @@
+/**
+ * @file genome.hpp
+ * @author Bang Liu
+ *
+ * Definition of the Genome class.
+ */
+#ifndef MLPACK_METHODS_NE_GENOME_HPP
+#define MLPACK_METHODS_NE_GENOME_HPP
+
+#include <cstddef>
+#include <cassert>
+#include <map>
+
+#include <mlpack/core.hpp>
+
+#include "link_gene.hpp"
+#include "neuron_gene.hpp"
+#include "utils.hpp"
+
+namespace mlpack {
+namespace ne {
+
+/**
+ * This class defines a genome.
+ # A genome is consist of a group of genes.
+ */
+class Genome {
+ public:
+  // Neurons.
+  std::vector<NeuronGene> aNeuronGenes;
+
+  // Links.
+  std::vector<LinkGene> aLinkGenes;
+
+  // Default constructor.
+  Genome() {
+    aId = -1;
+    aNumInput = 0;
+    aNumOutput = 0;
+    aFitness = DBL_MAX;
+  }
+  
+  // Parametric constructor.
+  Genome(int id,
+  	     const std::vector<NeuronGene>& neuronGenes,
+         const std::vector<LinkGene>& linkGenes,
+         int numInput,
```

Comment:
I think it shall make no difference. Besides, it will make a lot of consts.
http://stackoverflow.com/questions/117293/use-of-const-for-function-parameters